### DISCUSSION TITLE
Run Aggregator in Kubernetes

### DISCUSSION THREAD

**Discussion by DKondratyev** [tfsaggregator/aggregator-cli#256] [Q&A]
Hi guys,

currently I am trying to connect Azure DevOps 2020 (on-premise) with Aggregator 3 running as a pod in our Kubernetes cluster. 

The environment variables in the running container are set as follows:

![env](https://user-images.githubusercontent.com/93582746/157827123-4767cc18-82e0-4e26-960e-a4d34db1f6bc.png)

When an event is triggered in Azure DevOps, the aggregator detects it, but the connection from aggregator to Azure DevOps seems not to be established:

![attach](https://user-images.githubusercontent.com/93582746/157760800-e96c6331-1873-409a-8499-48f92b8979f6.png)

Do you have an idea what can go wrong ? Is it possible that the Azure DevOps' ssl certificate should also be placed somewhere in the container ?

Does the error "ApiKeyAuthenticationScheme was not authenticated. Failure message: Api Key is null" has something to do with the connection problem ? Why does it occur ? The file apikeys.json is available in /secrets folder ...

I would be thankful for any hints. :)

Have a nice evening !

Best regards,
Dmitry

**Comment by giuliov** [comment]
Sorry for the delay, I was offline this weekend. I will take a look at it soon.

**Reply by DKondratyev** [reply]
Hi Giulio, the problem is now solved. After the environment variable Aggregator_AzureDevOpsCertificate is set and the Azure DevOps' certificate is put into the /secrets folder, everything worked perfectly. :) Thank you !

**Reply by giuliov** [reply]
Great to hear!
Any suggestion to improve configuration of the documentation?

**Comment by DKondratyev** [comment]
Hi Giulio,
one short question regarding TFS Aggregator rules :) 
There is a function
public T GetFieldValue<T>(string field, T defaultValue)

which can be used in Aggregator's rule. The parameter "field" is the whole name. Suppose I need a list of all field values having some substring in the field name ( like "field name starts with WEF_"). How could I get it ? Is it also possible to get the list of all fields and their values in one dictionary from the self-object?
Thank you in advance !
Best regardsDmitry    Am Dienstag, 15. März 2022 um 07:57:47 MEZ hat Giulio Vian ***@***.***> Folgendes geschrieben:  
 
 


Great to hear!
Any suggestion to improve configuration of the documentation?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by giuliov** [reply]
No, although it won't be hard to do. Implementation can be like `WorkItemStore.GetBacklogWorkItemTypesAndStates`.

**Comment by DKondratyev** [comment]
The background of my question: I would like to move a card to a certain board column as a reaction of some user action. Simple
self["System.BoardColumn"] = "XYZ" 
doesn't work. But it seems to be possible to set the first WEF-field like

 self["WEF_D159030290514518A228357FD7D968FE_Kanban.Column"] = " XYZ" 

After that the card will move. The question is only how to access the WEF field ... 

    Am Freitag, 2. September 2022 um 10:37:11 MESZ hat Giulio Vian ***@***.***> Folgendes geschrieben:  
 
 


No, although it won't be hard to do. Implementation can be like GetBacklogWorkItemTypesAndStates.
